src/components/Checkout.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from "react";
+import { Table, Button } from "reactstrap";
+import { get,post } from "../clients/HttpClient";
+import { APP_PROPS } from "../constants/AppConstants";
+import { Link } from "react-router-dom";
+import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+
+function Checkout() {
+    const [cartData, setCartData] = useState([]);
+    const [cartTotal, setCartTotal] = useState(0);
+    const [dropdownOpen, setDropdownOpen] = useState(false);
+    const [selectedAddress, setSelectedAddress] = useState(null);
+    const [addresses, setAddresses] = useState([]);
+
+    const toggleDropdown = () => setDropdownOpen(prevState => !prevState);
+
+    const selectAddress = (address) => {
+        setSelectedAddress(address);
+        setDropdownOpen(false);
+    };
+
+    const clearSelection = () => {
+        setSelectedAddress(null);
+    };
+
+    const fetchAddresses = () => {
+        const addressUrl = `${APP_PROPS.bookstoreUrl}/address`;
+        get(addressUrl,
+            response => {
+                setAddresses(response.data);
+            },
+            error => {
+                console.error("Error fetching addresses:", error);
+            });
+    };
+
+    const fetchCartSummary = () => {
+        const cartSummaryUrl = `${APP_PROPS.bookstoreUrl}/cart/checkout`;
+        get(cartSummaryUrl,
+            response => {
+                setCartData(response.data.items);
+                setCartTotal(response.data.total);
+            },
+            error => {
+                console.error("Error fetching cart summary:", error);
+            });
+    };
+
+    useEffect(() => {
+        fetchAddresses();
+        fetchCartSummary();
+    }, []);
+
+    const createCartTableRow = (cartItem) => (
+        <tr key={cartItem.id}>
+            <td>{cartItem.id}</td>
+            <td>{cartItem.name}</td>
+            <td>${cartItem.price.toFixed(2)}</td>
+            <td>{cartItem.quantity}</td>
+            <td>${cartItem.total.toFixed(2)}</td>
+        </tr>
+    );
+
+    const handlePlaceOrder = async () => {
+        if (selectedAddress) {
+            const request = { id: selectedAddress.id };
```

Comment:
the /orders api takes key as addressId.
the /address api uses key as id.
{addressId: selectedAddress.id} is the correct approach